tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+        buckets_deleted = []
+        iam_deleted = []
+        for bucket in self.buckets_created:
+            resp = s3_misc.delete_objects_bucket(bucket[0], bucket[1], bucket[2])
+            if resp:
+                buckets_deleted.append(bucket)
+            else:
+                self.log.error("Bucket deletion failed for %s ", bucket)
+        self.log.info("buckets deleted %s", buckets_deleted)
+        for bucket in buckets_deleted:
+            self.buckets_created.remove(bucket)
+
+        self.log.info("Deleting iam account %s created in test", self.created_iam_users)
+        for iam_user in self.created_iam_users:
+            resp = self.csm_obj.delete_iam_user(iam_user)
+            if resp.status_code == HTTPStatus.OK:
+                iam_deleted.append(iam_user)
+            else:
+                self.log.error("IAM deletion failed for %s ", iam_user)
+        self.log.info("IAMs deleted %s", iam_deleted)
+        for iam in iam_deleted:
+            self.created_iam_users.remove(iam)
+        assert_utils.assert_true(len(self.buckets_created) == 0, "Bucket deletion failed")
+        assert_utils.assert_true(len(self.created_iam_users) == 0, "IAM deletion failed")
+        self.log.info("ENDED: Test teardown.")
+
+    def create_iam_user(self):
+        """
+        It will create IAM user and return s3test obj and s3multipart obj.
+        """
+        self.log.info("Creating S3 account and IAM user")
+        payload = self.csm_obj.iam_user_payload_rgw("random")
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert resp.status_code == HTTPStatus.CREATED, "IAM user creation failed"
+        user_id = resp.json()["keys"][0]['user']
+        self.created_iam_users.add(user_id)
+        resp1 = self.csm_obj.compare_iam_payload_response(resp, payload)
+        self.log.info("Printing response %s", resp1)
+        assert resp1[0], resp1[1]
+        akey = resp.json()["keys"][0]["access_key"]
+        skey = resp.json()["keys"][0]["secret_key"]
+        return akey, skey, user_id
+
+    def set_get_user_quota(self, config_dict, user_id):
+        """
+        It will create IAM user and return s3test obj and s3multipart obj.
+        """
+        self.log.info("Perform get set user quota")
+        payload = self.csm_obj.iam_user_quota_payload(config_dict["enabled"],
+                                                      config_dict["max_size"],
+                                                      config_dict["max_objects"],
+                                                      check_on_raw=True)
+        result, resp = self.csm_obj.verify_get_set_user_quota(user_id, payload,
+                                                              verify_response=True)
+        assert result, f"Verification for get set user failed with error {resp}"
+
+    def verify_user_quota(self, akey, skey, user_id):
```

Comment:
pls move this to s3_misc

Reply:
@AshwiniS-Borse please see replies to earlier discussion. Unresolving the discussion thread. cc: @archanabpatil